### PR TITLE
(DPS-2742) Migrate search component

### PR DESCRIPTION
**Pull request by dominikslawkowski**
## 🚀 Links

- [Feature branch](https://developers.labs.livechat.com/docs/feature/DPS-2742)
- [Jira](https://livechatinc.atlassian.net/browse/DPS-2742)

## ✅ Checklist (for API docs)

- Changelog
- API versions
- Web & RTM
- **Table:** Available methods/webhooks/pushes
- **Table:** Specifics

For more guidelines, see [Updating API docs](https://livechatinc.atlassian.net/wiki/spaces/PAT/pages/761529400/Updating+API+docs).

### Content

- New content
- Proofreading/content fixes
  
### Features (for the website)

- New feature
- Bug fix
- Breaking change

### CODE REVIEW

File: src/components/core/SideNav/SideNav.js [hunk 1]
```diff
@@ -1,4 +1,7 @@
 import React, { useState, Fragment, useContext } from "react";
+import Link from "next/link";
+import dynamic from "next/dynamic";
```
Comment (jakubsikora) [inline src/components/core/SideNav/SideNav.js:3]: Do you need that ?

File: src/pages/_app.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import "docsearch.js/dist/cdn/docsearch.min.css";
```
Comment (jakubsikora) [inline src/pages/_app.js:1]: I dont think you need that. We are adding that in the head already, but if that works, perhaps you can remove inserting css in the seo.js component.

File: src/utils/index.js [hunk 3]
```diff
@@ -78,3 +45,33 @@ export const setUrlParams = (section) => {
...
+export const setupDocsearch = () => {
+  if (!canUseWindow) {
+    return;
+  }
+
+  window.docsearch({
+    apiKey: process.env.NEXT_PUBLIC_ALGOLIA_API_KEY,
+    indexName: "livechatinc",
+    inputSelector: "#search",
+    debug: true, //false,
```
Comment (jakubsikora) [inline src/utils/index.js:58]: should be false

**Review by jakubsikora — CHANGES_REQUESTED** [review]
Cool! Few comments.

File: src/components/core/SideNav/SideNav.js [hunk 4]
```diff
@@ -87,16 +89,15 @@ const SideNav = ({
...
+  // const categoryMeta = useCategoryMeta(category);
+  // useScrollSpy(".heading", (url) => url && setActivePath(getArticlePath(url)));
 
   const navColor = getVersionColor(selectedVersion, versions);
 
   return (
     <Nav color={navColor} expanded={expanded} setExpanded={setExpanded}>
       <NavHeader>
-        <Link to={"/"} style={{ color: "inherit" }}>
+        <Link href="/" style={{ color: "inherit" }}>
```
Comment (arturfracala) [inline src/components/core/SideNav/SideNav.js:99]: It's missing `<a>` tag here. Look at documentation -> https://nextjs.org/docs/migrating/from-gatsby#links

**Review by arturfracala — APPROVED** [review]
Apart from the things that Jakub pointed out, it looks good to me. Nice job! 😄 🎉

**Review by jakubsikora — APPROVED** [review]

**Pull request merged**